stdlib/http/src/main/java/org/ballerinalang/net/http/serviceendpoint/Register.java

```diff
@@ -66,14 +66,17 @@ public void execute(Context context) {
         HTTPServicesRegistry httpServicesRegistry = getHttpServicesRegistry(serviceEndpoint);
         WebSocketServicesRegistry webSocketServicesRegistry = getWebSocketServicesRegistry(serviceEndpoint);
 
-        TypeRefCPEntry listenerType = service.getServiceInfo().listenerType;
-        if (listenerType == null || HttpConstants.HTTP_SERVICE_ENDPOINT_NAME.equals(listenerType.getTypeSig())) {
-            httpServicesRegistry.registerService(service);
-        } else if (WebSocketConstants.WEBSOCKET_ENDPOINT_NAME.equals(listenerType.getTypeSig())) {
-            WebSocketService webSocketService = new WebSocketService(service);
-            webSocketServicesRegistry.registerService(webSocketService);
+        ParamDetail param = service.getResources()[0].getParamDetails().get(0);
+        if (param != null) {
+            String listenerType = param.getVarType().toString();
```

Comment:
Shall we rename the `listenerType` variable as well since it is not the listener but the caller type that we check here